### DISCUSSION TITLE
Beta function

### DISCUSSION THREAD

**Discussion by thedailygrinder** [markrogoyski/math-php#444] [Q&A]
Hello guys,

I'm trying to run the Beta function as described in the documentation:

// Beta function
[$x, $y] = [1, 2];
$β  = Special::beta($x, $y);
$lβ = Special::logBeta($x, $y);

However, it appears that the function is not accepting inputs as strings. Are there any specific nuances to be aware of or is this a bug?

Many thanks!

**Comment by Beakerboy** [comment]
Thanks for getting in touch. What error are you receiving? Also, I’m confused when you say the inputs are strings. In your example `$x` and `$y` are holding integers.

Please provide as much context as is needed to reproduce your issue, like a complete function, CLI statement, or php file.

The unit testing suite includes your specific example:
```
MathPHP\Tests\Functions\SpecialTest::testBeta

[$x, $y, $result] = [1, 2, 0.5];
```

**Comment by markrogoyski** [comment]
Hi @thedailygrinder,

The `beta` function has typed arguments `$a` and `$b` that are of type `float`.

```php
public static function beta(float $a, float $b): float
```

PHP has loose typing, so if you pass a number as a string, PHP will coerce the value into a number and work like you had passed in a number type. However, if the calling code is in a file that declared `strict_types` to be on, then passing a string will fail the type declaration and produce a `TypeError`.

It sounds like you have declared `strict_types` in your code. If you want to be able to coerce strings into numbers when calling functions with scalar type declarations like those found in MathPHP, I suggest removing this line from the top of your calling PHP file:

```php
declare(strict_types=1);
``` 

Mark

**Reply by Beakerboy** [reply]
TIL…I’ve never used strict_types. Happy New Years Mark!

**Comment by markrogoyski** [comment]
Happy New Year!

Yeah, strict types is a bit unintuitive. You can force your own code to respect scalar types in code it calls, but you can't force other code to respect your own types when it calls your functions.
